.github/workflows/build.yml

```diff
@@ -115,9 +172,136 @@ jobs:
         CIBW_TEST_COMMAND: py.test -v {project}/tests
 
     - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v3
+      with:
+        name: packages
+        path: ./dist
+
+  build-sdist:
+    name: sdist - ${{ matrix.os.name }} ${{ matrix.python.major-dot-minor }} ${{ matrix.arch.name }}
+    runs-on: ${{ matrix.os.runs-on[matrix.arch.matrix] }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - name: Ubuntu
+            matrix: ubuntu
+            runs-on:
+              arm: [Linux, ARM64]
```

Comment:
Why do we have both listed here if we only really need to build this once? (It only appears to be running once, but seems like the sdist step could maybe even be simplified to not use matrix syntax at all, and just run straight on ubuntu-latest or something like that?)

Reply:
Both things are for copy-pasteability, and yes they come with a cost.  I think maybe what I want is for GitHub Actions to support yaml anchors, then I would put a 'library' of things generated into the bottom of every workflow and would reference those instead of copy/pasting all over.  CI runs would make sure the 'library' was up to date.  Anyways, with this we can copy paste the setup python step, job name info, etc around without editing each time.  But yeah, this isn't the heart of the change here and is kind of ugly.